src/utils/table.tsx

```diff
@@ -0,0 +1,110 @@
+import { useMemo } from "react";
+import {
+  MaterialReactTable,
+  useMaterialReactTable,
+  type MRT_ColumnDef,
+} from "material-react-table";
```

Comment:
Separate the column definitions and table implementation into different files.
 This improves code readability and maintainability by keeping concerns isolated:
 columns.ts: Defines the structure and configuration of the table columns.
 TableComponent.tsx: Implements the Material React Table using the defined columns.
 This structure makes it easier to manage and update the table's configuration and logic independently.